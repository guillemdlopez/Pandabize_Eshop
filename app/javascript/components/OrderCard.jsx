import React from "react";

const OrderCard = ({ customizations, bicycle, photo, amount }) => {
  return (
    <div className="order-card order-card-dashboard">
      <img src={photo} className="dashboard-card-img" />
      <div className="dashboard-card-content">
        <ul className="block-list-left ml-3 mb-0">
          <li>
            <strong>Bicycle:</strong> {bicycle}
          </li>
          <li>
            <strong>Amount:</strong> {amount} â‚¬
          </li>
          {customizations.map((custom) => {
            return (
              <li key={custom.id} className="custom-item mt-2">
                <strong>{custom.name}:</strong> {custom.value}
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  );
};

export default OrderCard;
