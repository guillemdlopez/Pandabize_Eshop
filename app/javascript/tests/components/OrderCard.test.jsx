import { shallow } from "enzyme";
import React from "react";
import OrderCard from "../../components/OrderCard";

describe("Testing the <OrderCard /> component", () => {
  const order = {
    id: 19,
    bicycle_id: 26,
    amount: 13.1,
    user_id: 10,
    bicycle: "New School XC",
    photo:
      "http://res.cloudinary.com/doker55/image/upload/jbwjj6i6r1f2oz551vxojkmtb42c.jpg",
    customizations: [
      {
        id: 148,
        name: "Wheel size",
        value: "16 inches",
        price: 6.15,
        bicycle_id: 26,
      },
      {
        id: 168,
        name: "Rim color",
        value: "Blue",
        price: 6.95,
        bicycle_id: 26,
      },
    ],
  };
  const wrapper = shallow(<OrderCard {...order} />);

  test("should render the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should display the name of the bicycle", () => {
    const { bicycle } = order;
    expect(wrapper.find("li").at(0).text()).toBe(`Bicycle: ${bicycle}`);
  });

  test("should display the amount of the bicycle", () => {
    const { amount } = order;
    expect(wrapper.find("li").at(1).text()).toBe(`Amount: ${amount} â‚¬`);
  });

  test("should display the customizations names and their values", () => {
    const value = order.customizations.map((custom) => custom.value);
    const name = order.customizations.map((custom) => custom.name);

    expect(wrapper.find(".custom-item").at(0).text()).toBe(
      `${name[0]}: ${value[0]}`
    );
    expect(wrapper.find(".custom-item").at(1).text()).toBe(
      `${name[1]}: ${value[1]}`
    );
  });
});
