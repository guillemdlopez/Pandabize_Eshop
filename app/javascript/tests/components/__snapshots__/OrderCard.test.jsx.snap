// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <OrderCard /> component should render the component correctly 1`] = `
<div
  className="order-card order-card-dashboard"
>
  <img
    className="dashboard-card-img"
    src="http://res.cloudinary.com/doker55/image/upload/jbwjj6i6r1f2oz551vxojkmtb42c.jpg"
  />
  <div
    className="dashboard-card-content"
  >
    <ul
      className="block-list-left ml-3 mb-0"
    >
      <li>
        <strong>
          Bicycle:
        </strong>
         
        New School XC
      </li>
      <li>
        <strong>
          Amount:
        </strong>
         
        13.1
         â‚¬
      </li>
      <li
        className="custom-item mt-2"
        key="148"
      >
        <strong>
          Wheel size
          :
        </strong>
         
        16 inches
      </li>
      <li
        className="custom-item mt-2"
        key="168"
      >
        <strong>
          Rim color
          :
        </strong>
         
        Blue
      </li>
    </ul>
  </div>
</div>
`;
